### ISSUE TITLE
System hook push event  autor's id and username field are both null ?

### ISSUE THREAD

**Issue by WestFarmer** [gitlab4j/gitlab4j-api#402]
```
Jan 19 06:12:13 ocrpoc1 java: 2003-01-19 06:12:13.222  INFO 21729 --- [nio-9876-exec-9] c.b.d.d.g.h.l.DefaultSystemHookListener  : ====== Push Event ======
Jan 19 06:12:13 ocrpoc1 java: 2003-01-19 06:12:13.222  INFO 21729 --- [nio-9876-exec-9] c.b.d.d.g.h.l.DefaultSystemHookListener  : {
Jan 19 06:12:13 ocrpoc1 java: "id" : "b5f6a08bba2db1734004b309a456f2799ba36848",
Jan 19 06:12:13 ocrpoc1 java: "message" : "test push event",
Jan 19 06:12:13 ocrpoc1 java: "timestamp" : "2019-07-08T09:10:16Z",
Jan 19 06:12:13 ocrpoc1 java: "url" : "http://172.31.1.195/public_group/dtis/commit/b5f6a08bba2db1734004b309a456f2799ba36848",
Jan 19 06:12:13 ocrpoc1 java: "author" : {
Jan 19 06:12:13 ocrpoc1 java: "avatarUrl" : null,
Jan 19 06:12:13 ocrpoc1 java: "createdAt" : null,
Jan 19 06:12:13 ocrpoc1 java: "email" : "fanguogang@bgzchina.com",
Jan 19 06:12:13 ocrpoc1 java: "id" : null,
Jan 19 06:12:13 ocrpoc1 java: "name" : "范国刚",
Jan 19 06:12:13 ocrpoc1 java: "state" : null,
Jan 19 06:12:13 ocrpoc1 java: "username" : null,
Jan 19 06:12:13 ocrpoc1 java: "webUrl" : null
Jan 19 06:12:13 ocrpoc1 java: },
Jan 19 06:12:13 ocrpoc1 java: "added" : [ ],
Jan 19 06:12:13 ocrpoc1 java: "modified" : [ "src/main/java/com/bgzchina/devops/dtis/gitlab/hook/listeners/DefaultSystemHookListener.java" ],
Jan 19 06:12:13 ocrpoc1 java: "removed" : [ ]
Jan 19 06:12:13 ocrpoc1 java: }
```

**Comment by gmessner** [comment]
@WestFarmer 
A couple of questions:
1) What version of GitLab is being used? 
2) What is the source of that event?  It looks nothing like it should, see  https://docs.gitlab.com/ee/system_hooks/system_hooks.html#push-events

**Comment by gmessner** [comment]
@WestFarmer 
My apologies, I can now see that the log is for a single commit from the event and does, in fact, look correct.

To answer your question, the ```id``` and ```username``` fields are not included in the event JSON from the server, so they will be null.  If you dump the actual JSON message received you will see that those fields are not included in the message.  You can also check the documentation which also shows that they are not included (https://docs.gitlab.com/ee/system_hooks/system_hooks.html#push-events).

The reason you see ```id``` and ```username``` in your log is that the ```Author``` class includes those fields as well as a lot of other fields.

Basically, the answer to your question is that this is the expected behavior.

**Comment by WestFarmer** [comment]
@gmessner 
Thanks for your help. I got it.
I think I understand why that happens:
I guess when people commit, only user's name and email is sent to GitLab server, that's what you can set with `git config`, and in gitlab's user settings, there is field called "commit email", we can use it to find user by call user api.

**Comment by gmessner** [comment]
@WestFarmer 
Yes, you are correct, you are seeing the info that is normally included in a commit, the other info like ```id``` and ```username``` are sensitive information and the receiver of the event may not have sufficient privileges to see that inf. If the receiver does have privileges to see that info it can use the ```UserApi``` to load the user.

It would be nice if there was a method in the ```UserApi``` that would return an ```Optional<User>``` based on email, I'll add an issue to add these methods:
```
public User getUserByEmail(String email) throws GitLabApiException
public Optional<User> getOptionalUserByEmail(String email)
```
The email will be validated prior to making the actual REST call.

**Comment by gmessner** [comment]
@WestFarmer 
GitLab4J-API Release 4.11.16 is now available and it includes the 2 new methods mentioned above.
